src/objective-c/GRPCClient/private/GRPCCore/GRPCHost.m

```diff
@@ -113,20 +113,10 @@ - (GRPCCallOptions *)callOptions {
   options.PEMPrivateKey = _PEMPrivateKey;
   options.PEMCertificateChain = _PEMCertificateChain;
   options.hostNameOverride = _hostNameOverride;
-#ifdef GRPC_COMPILE_WITH_CRONET
-  // By old API logic, insecure channel precedes Cronet channel; Cronet channel preceeds default
-  // channel.
-  if ([GRPCCall isUsingCronet]) {
-    if (_transportType == GRPCTransportTypeInsecure) {
-      options.transportType = GRPCTransportTypeInsecure;
-    } else {
-      NSAssert(_transportType == GRPCTransportTypeDefault, @"Invalid transport type");
-      options.transportType = GRPCTransportTypeCronet;
-    }
-  } else
-#endif
-  {
-    options.transportType = _transportType;
+  if (_transportType == GRPCTransportTypeInsecure) {
+    options.transport = GRPCDefaultTransportImplList.core_insecure;
+  } else {
+    options.transport = GRPCDefaultTransportImplList.core_secure;
```

Comment:
We cannot query `[GRPCCall isUsingCronet]` at this place because, if an SDK user is using v2 API with `transportType = GRPCTransportTypeDefault` and another SDK sets useCronet, this user will pick up Cronet automatically, which is against the design philosophy of v2 API (I believe it was a bug here). 

However, there's a good distinction between v1 users and v2 users. If a user uses v2 API, `GRPCCall2` makes sure that a default GRPCCallOptions object is constructed before it passes requests to lower layers (because interceptors need to act on an actual GRPCCallOptions, not nil). On the other hand, for v1 users they just don't have a call options object. So I moved the v1+Cronet logic [here](https://github.com/grpc/grpc/blob/25432e011a0bf4ae8f48ad16f251aeab89966ca0/src/objective-c/GRPCClient/GRPCCallLegacy.m#L592).

I'll make some comments in the code too. Hope this makes sense.

Reply:
Discard the previous comment.

I moved Cronet check to [here](https://github.com/grpc/grpc/blob/25432e011a0bf4ae8f48ad16f251aeab89966ca0/src/objective-c/GRPCClient/GRPCCallLegacy.m#L592). But as I look at it I think it's probably ok to put it in GRPCHost.m.